### PR TITLE
fix: Use provider source

### PR DESCRIPTION
**Pull request by zagronitay**

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -49,8 +51,14 @@ func Initialize(ctx context.Context, providers []string) error {
 	rootBody := f.Body()
 	requiredProviders := make([]*config.RequiredProvider, len(providers))
 	for i, p := range providers {
+		organization, providerName, err := registry.ParseProviderName(p)
+		if err != nil {
+			return fmt.Errorf("cloud not parse requested provider")
```
Comment (disq) [inline cmd/init.go:56]:
```suggestion
			return fmt.Errorf("could not parse requested provider")
```

**Review by disq — COMMENTED** [review]
LGTM

File: pkg/plugin/registry/hub.go [hunk 2]
```diff
@@ -146,7 +146,13 @@ func (h Hub) VerifyProvider(ctx context.Context, organization, providerName, ver
...
 // CheckProviderUpdate - checks if there is an update for provider, returns nil if there is no update
 func (h Hub) CheckProviderUpdate(ctx context.Context, requestedProvider *config.RequiredProvider) (*string, error) {
-	organization, providerName, err := ParseProviderName(requestedProvider.Name)
+	var requestedSource string
+	if requestedProvider.Source == nil {
+		requestedSource = requestedProvider.Name
+	} else {
+		requestedSource = *requestedProvider.Source
+	}
+	organization, providerName, err := ParseProviderName(requestedSource)
```
Comment (roneli) [inline pkg/plugin/registry/hub.go:155]: nit: make this a function of h.Hub

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, minor nit comment

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**